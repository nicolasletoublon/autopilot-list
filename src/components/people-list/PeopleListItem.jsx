import React from 'react';
import styled from 'styled-components';
import SvgIcon from '../svg-icon';
import PropTypes from 'prop-types';
import Adornment from '../adornment/Adornment';

const ListItemWrapper = styled.li`
  display: flex;
  align-items: center;
  padding: 10px 20px 10px 0;
  color: #343434;
  height: 40px;

  &:hover {
    background: #e8fffa;

    ${SvgIcon} {
      fill: #35e6c3;
    }
  }
`;

const AdornmentWrapper = styled.div`
  display: flex;
  justify-content: center;
  flex: 20%;
`;

const NameWrapper = styled.div`
  flex: 80%;
`;
NameWrapper.displayName = 'NameWrapper';

const PeopleListItem = ({ name, avatar }) => {
  return (
    <ListItemWrapper>
      <AdornmentWrapper>
        <Adornment avatar={avatar} name={name} />
      </AdornmentWrapper>
      <NameWrapper>{name}</NameWrapper>
    </ListItemWrapper>
  );
};

PeopleListItem.defaultProps = {
  avatar: undefined,
};

PeopleListItem.propTypes = {
  avatar: PropTypes.string,
  name: PropTypes.string.isRequired,
};

PeopleListItem.displayName = 'PeopleListItem';

export default PeopleListItem;
