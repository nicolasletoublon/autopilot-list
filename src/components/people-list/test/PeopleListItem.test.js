import React from 'react';
import { shallow, mount } from 'enzyme';
import PeopleListItem from '../PeopleListItem';

describe('PeopleListItem', () => {
  describe('should render correctly the name and the adornment', () => {
    const comp = <PeopleListItem name="Name" />;

    it('snapshot', () => {
      const component = shallow(comp);
      expect(component).toMatchSnapshot();
    });

    it('should find the right components in the DOM with the right props', () => {
      const component = mount(comp);
      expect(component.find('Adornment')).toHaveLength(1);
      expect(component.find('Adornment').props().name).toEqual('Name');
      expect(component.find('Adornment').props().avatar).toBeUndefined();
      expect(component.find('NameWrapper')).toHaveLength(1);
      expect(component.find('NameWrapper').props().children).toEqual('Name');
    });
  });
});
