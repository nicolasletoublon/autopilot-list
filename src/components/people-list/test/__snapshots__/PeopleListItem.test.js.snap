// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PeopleListItem should render correctly the name and the adornment snapshot 1`] = `
<styled.li>
  <styled.div>
    <Adornment
      name="Name"
    />
  </styled.div>
  <NameWrapper>
    Name
  </NameWrapper>
</styled.li>
`;
